### ISSUE TITLE
Problems getting hello_world running on ESP-IDF 4.0

### ISSUE THREAD

**Issue by zdila** [manuelbl/ttn-esp32#14]
For ESP-IDF I had to perform following changes for hello_world example to compile:

- add `#include <freertos/semphr.h>` to `hal_esp32.h`
- add `#include "driver/gpio.h"` to `main.cpp`

Thanks for awesome library!

**Comment by manuelbl** [comment]
Thanks for the info.

Most likely I will not upgrade the library to ESP-IDF 4.0 until it's officially released. But with your changes, anyone can upgrade earlier.

**Comment by zdila** [comment]
Maybe a macro could be used now:
```c
#if IDF_MAJOR_VER >= 4
#include <freertos/semphr.h>
#endif
```

```c
#if IDF_MAJOR_VER >= 4
#include "driver/gpio.h"
#endif
```

**Comment by manuelbl** [comment]
ESP-IDF 4.0 is not stable yet. So any such code change might not work anymore with later development releases or with the final 4.0 release. I'm very reluctant to implement it.

**Comment by zdila** [comment]
OK, no problem, just giving some hints ;-)

**Comment by manuelbl** [comment]
The *dev* branch now contains a major update to the library including compatibility with the upcoming version 4 of ESP-IDF.

**Comment by manuelbl** [comment]
Release 3.2.0 is has been released and is based on ESP-IDF 4.0.x
